core/src/test/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitorSuite.scala

```diff
@@ -333,6 +333,19 @@ class ExecutorMonitorSuite extends SparkFunSuite {
     assert(monitor.timedOutExecutors(idleDeadline) === Seq("1"))
   }
 
+
+  test("SPARK-28839: Avoids NPE in context cleaner when shuffle service is on") {
+    val bus = mockListenerBus()
+    conf.set(DYN_ALLOCATION_SHUFFLE_TRACKING, true).set(SHUFFLE_SERVICE_ENABLED, true)
+    monitor = new ExecutorMonitor(conf, client, bus, clock)
+    monitor.onExecutorAdded(SparkListenerExecutorAdded(clock.getTimeMillis(), "1", null))
+    try {
+      monitor.shuffleCleaned(0)
+    } catch {
+      case _: NullPointerException => fail(s"NullPointerException was thrown.")
```

Comment:
You don't need the try...catch block. it's actually more helpful to see the original exception.